featurebyte/utils/async_helper.py

```diff
@@ -173,31 +171,16 @@ async def _semaphore_refresh_task() -> None:
                 await self._refresh()
 
         # Try to acquire semaphore
-        now = time.time()
-        while time.time() - now < self._timeout:
-            # Acquire lock for semaphore
-            lock = self._redis.lock(self._lock, timeout=10)
-            if lock.acquire(blocking=False):
-                try:
-                    identifier = await self._acquire_semaphore()
-                    if identifier:
-                        self._identifier = identifier
-                        # Add task to refresh semaphore while semaphore is acquired
-                        self._semaphore_refresh_task = asyncio.create_task(
-                            _semaphore_refresh_task()
-                        )
-                        return self
-                finally:
-                    # Release lock
-                    if lock.owned():
-                        try:
-                            lock.release()
-                        except LockNotOwnedError:
-                            # Lock may have expired before release
-                            pass
-            await asyncio.sleep(0.1)
-        # Timeout trying to acquire semaphore
-        raise TimeoutError(f"Failed to acquire semaphore after {time.time() - now}s")
+        async with acquire_lock(
```

Comment:
@kchua78 I refactored the async lock acquisition logic in `RedisFairCountingSemaphore` to make it reusable. Let me know if you see any issues.

Reply:
`self._acquire_semaphore` is non-blocking and fails immediately when not successful, so we will still need the loop to keep trying within the timeout

Reply:
Indeed. Will revert these changes for now to minimize risk.